dojo/engagement/views.py

```diff
@@ -569,45 +586,32 @@ def import_scan_results(request, eid=None, pid=None):
 
-            if not any(scan_type in code
-                       for code in ImportScanForm.SORTED_SCAN_TYPE_CHOICES):
-                raise Http404()
-            if file and is_scan_file_too_large(file):
-                messages.add_message(request,
-                                     messages.ERROR,
-                                     "Report file is too large. Maximum supported size is {} MB".format(settings.SCAN_FILE_MAX_SIZE),
-                                     extra_tags='alert-danger')
-                return HttpResponseRedirect(reverse('import_scan_results', args=(engagement,)))
+            # can't use helper as when push_all_jira_issues is True, the checkbox gets disabled and is always false
+            # push_to_jira = jira_helper.is_push_to_jira(new_finding, jform.cleaned_data.get('push_to_jira'))
+            push_to_jira = push_all_jira_issues or (jform and jform.cleaned_data.get('push_to_jira'))
+            error = False
 
-            tt, t_created = Test_Type.objects.get_or_create(name=scan_type)
+            try:
+                importer = Importer()
+                test, finding_count, closed_finding_count = importer.import_scan(scan, scan_type, engagement, user, environment, active=active, verified=verified, tags=tags,
+                            minimum_severity=minimum_severity, endpoints_to_add=form.cleaned_data['endpoints'], scan_date=scan_date,
+                            version=version, branch_tag=branch_tag, build_id=build_id, commit_hash=commit_hash, push_to_jira=push_to_jira,
+                            close_old_findings=close_old_findings)
 
-            # Will save in the provided environment or in the `Development` one if absent
-            environment_id = request.POST.get('environment', 'Development')
-            environment = Development_Environment.objects.get(id=environment_id)
+                message = scan_type + '%s processed a total of %s findings' % (scan_type, finding_count)
+
+                if close_old_findings:
+                    message = message + ' and closed %d findings' % (closed_finding_count)
+
+                message = message + "."
 
-            t = Test(
-                engagement=engagement,
-                test_type=tt,
-                target_start=scan_date,
-                target_end=scan_date,
-                environment=environment,
-                percent_complete=100,
-                version=version,
-                branch_tag=branch_tag,
-                build_id=build_id,
-                commit_hash=commit_hash,
-                tags=tags)
-            t.lead = user
-            t.full_clean()
-            t.save()
+                add_success_message_to_response(message)
+
+            except Exception as e:
+                # exceptions are already logged by the importer
+                add_error_message_to_response('An exception error occurred during the report import:%s' % str(e))
+                error = True
```

Comment:
the way exception are managed is weird. catch everything is never the best solution. :/ also if we really want to log the exception, I prefer this form:

```Python
except XxxxxError as ex:
    logger.exception("something bad happened during import bla bla bla")
```

Reply:
that is exactly what happens one level deeper. but here we catch the exception to show a nice message in the UI. In api we expose the exception, that's why it all looks weird. but it's all existing code just moved somewhere else. people are free to improve further in future prs.

Reply:
ok. as discussed in Slack, if error management is part of "normal" workflow, we should return a boolean with the result that make explicit if the import was a success or not instead of this exception programming. but as you said it's all existing code just moved. we can improve it in further PRs